rustls/src/crypto/mod.rs

```diff
@@ -186,6 +206,63 @@ pub struct CryptoProvider {
     pub key_provider: &'static dyn KeyProvider,
 }
 
+impl CryptoProvider {
+    /// Sets the supplied `CryptoProvider` as the default for this process.
+    ///
+    /// This can be called successfully at most once in any process execution.
+    ///
+    /// Call this early in your process to configure which provider is used for
+    /// the provider.  The configuration should happen before any use of
+    /// [`ClientConfig::builder()`] or [`ServerConfig::builder()`].
+    pub fn install_as_process_default(provider: impl Into<Arc<Self>>) -> Result<(), Arc<Self>> {
+        PROCESS_DEFAULT_PROVIDER.set(provider.into())
+    }
+
+    /// Sets the default provider to the one named unambiguously by rustls crate features.
+    ///
+    /// This function does nothing if the crate features are ambiguous (ie, specify two
+    /// providers), or specify no providers.  In both cases the application should
+    /// explicitly specify the provider to use with [`CryptoProvider::install_as_process_default`].
+    pub fn install_process_default_from_crate_features() {
```

Comment:
Personally I find `install_as_process_default()` pretty wordy already, but this is really out there in terms of number of tokens. Is there no more compact version of this that would convey enough context? `set_default_from_feature()`? (Is it actually multiple features or rather a single feature?)

Reply:
I've split this up into `CryptoProvider::from_crate_features()`, stopped it doing the installing too, and made it private.

Reply:
That makes the public interface much better! I'd probably still prefer a more concise name even for internal API, but obviously that's much less important.